### PR TITLE
Add Typescript declarations

### LINKED ISSUE #2: Typescript Declarations
Since this repo monkey-patches `extends`, it'd be great to go ahead and export typing information to [DefinitelyTyped](https://github.com/DefinitelyTyped/DefinitelyTyped). Could be similar to this:

```
/// <reference types="jest" />

declare namespace jest {
  interface Matchers<R> {
    toMatchFile: (filename: string) => void
  }
   interface Expect {
    toMatchFile: (filename: string) => void
  }
}
```

see, f.e., https://github.com/jest-community/jest-extended/blob/master/types/index.d.ts

### PR DESCRIPTION
**Pull request by colinking**
This PR adds a Typescript declaration file, to resolve https://github.com/satya164/jest-file-snapshot/issues/2

### CODE REVIEW

File: types/index.d.ts [hunk 1]
```diff
@@ -0,0 +1,16 @@
...
+}
+
+declare namespace jest {
+  interface Matchers<R> {
+    toMatchFile: (output: string, filename: string, options?: FileMatcherOptions) => void
+  }
+   interface Expect {
+    toMatchFile: (output: string, filename: string, options?: FileMatcherOptions) => void
+  }
+}
```
Comment (Esemesek) [inline types/index.d.ts]: Add EOF. C:

**Review by Esemesek — CHANGES_REQUESTED** [review]
In addition to adding EOF, you should also link those types inside `package.json`.

**Comment by satya164** [comment]
Hey @colinking ! Would be awesome if you can address @Esemesek's review :)

File: types/index.d.ts [hunk 2]
```diff
@@ -0,0 +1,16 @@
...
+
+declare type FileMatcherOptions = {
+  diff: DiffOptions
+}
+
+declare namespace jest {
+  interface Matchers<R> {
+    toMatchFile: (output: string, filename: string, options?: FileMatcherOptions) => void
+  }
+   interface Expect {
```
Comment (satya164) [inline types/index.d.ts]: nit: wrong indentation

**Comment by colinking** [comment]
Updated the PR with your feedback. Thanks! 😄

**Comment by satya164** [comment]
Merged in 392dcd1368ecab296473455329feea17c6aa8492

**Pull request closed without merging**